java.util.concurrent.ExecutionException: java.io.IOException: Stream closed

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
RustRover 2024.3.2 (build RR-243.22562.230)

### Operating System 
Mac OS X 15.1.1 (aarch64)

### TeXiFy IDEA version
0.9.9

### Description



### Stacktrace
```
java.util.concurrent.ExecutionException: java.io.IOException: Stream closed
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at org.jetbrains.concurrency.AsyncPromise.get(AsyncPromise.kt:52)
	at org.jetbrains.concurrency.AsyncPromise.blockingGet(AsyncPromise.kt:130)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.executeSynchronously(NonBlockingReadActionImpl.java:513)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl.executeSynchronously(NonBlockingReadActionImpl.java:222)
	at com.intellij.util.indexing.roots.IndexableFilesIndexImpl.getIndexingIterators(IndexableFilesIndexImpl.java:78)
	at com.intellij.util.indexing.FileBasedIndexEx.getIndexableFilesProviders(FileBasedIndexEx.java:547)
	at com.intellij.util.indexing.projectFilter.ProjectIndexableFilesFilterHealthCheck.getFilesThatShouldBeIndexable(ProjectIndexableFilesFilterHealthCheck.kt:220)
	at com.intellij.util.indexing.projectFilter.ProjectIndexableFilesFilterHealthCheck.getFilesThatShouldBeIndexable(ProjectIndexableFilesFilterHealthCheck.kt:214)
        (...)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.ensureOpen(BufferedInputStream.java:206)
        (...)
	at kotlin.io.TextStreamsKt.readText(ReadWrite.kt:112)
	at nl.hannahsten.texifyidea.util.CommandRunnerKt$runCommandNonBlocking$2$output$1.invokeSuspend(CommandRunner.kt:54)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
```

--- a/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
+++ b/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
@@ -4,6 +4,7 @@ import com.intellij.execution.ExecutionException
 import com.intellij.execution.configurations.GeneralCommandLine
 import com.intellij.util.io.awaitExit
 import kotlinx.coroutines.*
+import java.io.BufferedReader
 import java.io.File
 import java.io.IOException
 import javax.swing.SwingUtilities
@@ -59,25 +60,8 @@ suspend fun runCommandNonBlocking(
         val process = processBuilder.start()
 
         process.outputWriter().use { if (input != null) it.write(input) }
-        val output = if (!discardOutput) async { process.inputReader().use { it.readText() } } else null
-        val error = if (!discardOutput) {
-            async {
-                process.errorReader().use {
-                    try {
-                        it.readText()
-                    }
-                    catch (e: IOException) {
-                        // In some cases directly after IDE start (after a timeout?), the stream may be closed already, so ignore that
-                        if (e.message?.contains("Stream closed") == true) {
-                            Log.info("Ignored closed stream: " + e.message)
-                            e.message
-                        }
-                        else throw e
-                    }
-                }
-            }
-        }
-        else null
+        val output = if (!discardOutput) async { process.inputReader().use { readTextIgnoreClosedStream(it) } } else null
+        val error = if (!discardOutput) { async { process.errorReader().use { readTextIgnoreClosedStream(it) } } } else null
 
         withTimeoutOrNull(1_000 * timeout) {
             process.awaitExit()
@@ -114,6 +98,18 @@ suspend fun runCommandNonBlocking(
             if (returnExceptionMessageAsErrorOutput) e.message else null
         )
     }
+}
+
+private fun readTextIgnoreClosedStream(reader: BufferedReader): String? = try {
+    reader.readText()
+}
+catch (e: IOException) {
+    // In some cases directly after IDE start (after a timeout?), the stream may be closed already, so ignore that
+    if (e.message?.contains("Stream closed") == true) {
+        Log.info("Ignored closed stream: " + e.message)
+        e.message
+    }
+    else throw e
 }
 
 /**

--- a/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
+++ b/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
@@ -61,7 +61,7 @@ suspend fun runCommandNonBlocking(
 
         process.outputWriter().use { if (input != null) it.write(input) }
         val output = if (!discardOutput) async { process.inputReader().use { readTextIgnoreClosedStream(it) } } else null
-        val error = if (!discardOutput) { async { process.errorReader().use { readTextIgnoreClosedStream(it) } } } else null
+        val error = if (!discardOutput) async { process.errorReader().use { readTextIgnoreClosedStream(it) } } else null
 
         withTimeoutOrNull(1_000 * timeout) {
             process.awaitExit()

